source/extensions/filters/http/ext_proc/processor_state.h

```diff
@@ -404,6 +422,13 @@ class EncodingProcessorState : public ProcessorState {
 
   Http::StreamFilterCallbacks* callbacks() const override { return encoder_callbacks_; }
 
+  bool sendAttributes(const ExpressionManager& mgr) const override {
+    return !attributes_sent_ && mgr.hasResponseExpr();
+  }
+
+  const Http::RequestOrResponseHeaderMap* requestHeaders() const override { return nullptr; };
```

Comment:
I believe the request headers should be available in the response phase. At least you can add a TODO comment here and optimize it in the future.

Reply:
This is a good catch. I think it would be prudent to have separate fields in the base class for request and response header maps to accomplish this, even though on decoding it will always be `nullptr` for the response headers.